### PR TITLE
archey4 4.11.0 (new formula)

### PR DESCRIPTION
**Pull request by HorlogeSkynet**
- [X] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [X] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [X] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [X] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [X] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

Thanks :wave:

> See HorlogeSkynet/archey4#96 for reference.

### CODE REVIEW

File: Formula/archey4.rb [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+    url "https://files.pythonhosted.org/packages/0d/18/fd6e9c71a35b67a73160ec80a49da63d1eed2d2055054cc2995714949132/netifaces-0.10.9.tar.gz"
+    sha256 "2dee9ffdd16292878336a58d04a20f0ffe95555465fee7c9bd23b3490ef2abf3"
+  end
+
+  def install
+    virtualenv_install_with_resources
+  end
+
+  test do
+    system "#{bin}/archey", "-v"
```
Comment (SMillerDev) [inline Formula/archey4.rb:27]: We need a test that exercises the some of the functionality of the app. Version checks or usage checks (foo --version or foo --help) are not sufficient, as explained in the formula cookbook.

In most cases, a good test would involve running a simple test case: run #{bin}/foo input.txt.

- Then you can check that the output is as expected (with assert_equal or assert_match on the output of shell_output)
- You can also check that an output file was created, if that is expected: assert_predicate testpath/"output.txt", :exist?

Some advice for specific cases:

- If the formula is a library, compile and run some simple code that links against it. It could be taken from upstream's documentation / source examples.
- If the formula is for a GUI program, try to find some function that runs as command-line only, like a format conversion, reading or displaying a config file, etc.
- If the software cannot function without credentials, a test could be to try to connect with invalid credentials (or without credentials) and confirm that it fails as expected.
- Same if the software requires a virtual machine, docker instance, etc. to be running.

Comment (HorlogeSkynet) [reply Formula/archey4.rb]: Thanks for your quick feedback @SMillerDev.
IMHO this is a _special_ case as we can't easily test one of the functionalities here.

There is no way I can guarantee you :

* a specific output (everything is system-dependent)
* a file would be created (only console output)

That's why I've just replaced `-v` flag by `-j`, which now implies a simple execution with minimal (relevant) output we can expect from it. If returned status code is zero, everything should be fine.

Bye :wave:

Comment (SMillerDev) [reply Formula/archey4.rb]: > a specific output (everything is system-dependent)

I'm not sure what information it provides, but you could check for things like "macOS|linux", "cpu", "directory" since those are gonna be possible in all hardware configurations.

Comment (HorlogeSkynet) [reply Formula/archey4.rb]: Please find a new patch implementing your proposal adapted for current supported platforms @SMillerDev.

**Comment by HorlogeSkynet** [comment]
Hello @SMillerDev, any news about last patch ? Thanks 🙇

File: Formula/archey4.rb [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+    url "https://files.pythonhosted.org/packages/0d/18/fd6e9c71a35b67a73160ec80a49da63d1eed2d2055054cc2995714949132/netifaces-0.10.9.tar.gz"
+    sha256 "2dee9ffdd16292878336a58d04a20f0ffe95555465fee7c9bd23b3490ef2abf3"
+  end
+
+  def install
+    virtualenv_install_with_resources
+  end
+
+  test do
+    ENV["DO_NOT_TRACK"] = "1"
```
Comment (SMillerDev) [inline Formula/archey4.rb:27]: Why is this needed?

Comment (HorlogeSkynet) [reply Formula/archey4.rb:27]: For privacy purposes, as running `archey -j` would automatically load internal `WAN_IP` entry performing outgoing requests. Setting this allows us to prevent this behavior without having to deal with configuration file.

Comment (SMillerDev) [reply Formula/archey4.rb:27]: I don't think our CI system has any privacy requirements. I'd say this can go.

Comment (HorlogeSkynet) [reply Formula/archey4.rb:27]: Oh I was actually thinking about end-users/developers tweaking formulae (or just this one for instance) and running `brew test <...>` on their own system, rather than the CI IP address.
What do you think about this eventual case ?

Comment (SMillerDev) [reply Formula/archey4.rb:27]: They probably have the tool installed and use it too. I don't think we should make the test more complicated than it needs to be.

Comment (HorlogeSkynet) [reply Formula/archey4.rb:27]: I don't want to bother you much but :

* Maybe they use this tool, but with this very variable exported in their shell profile :wink: 

* Sadly, I've just checked, `ENV` seems to be reset by `brew` when running tests (this means a developer exporting this variable in their profile would have no impact in this case)

* When coding Archey unit tests I particularly tried to take care of this issue, and I guess packaging/distributions scripts should try to do the same :roll_eyes:

* Final 2 cents : I don't think an environment variable complicates a test much, do it ?

Thanks for your time :wave:

Comment (HorlogeSkynet) [reply Formula/archey4.rb:27]: Hey @SMillerDev ! The boring guy is back. What do you think about my previous points ? Thanks, bye 👋

Comment (danielnachun) [reply Formula/archey4.rb:27]: I appreciate your concern here but I think the reality is that `brew test` is rarely used outside of CI, and when it is called manually it's usually done by maintainers.    

I don't have a strong feeling about this but I'm usually in favor of keeping tests as simple as possible.  There are certainly tests where we have to set some environment variables in order for them to work in CI, but this doesn't appear to be the case.  So I would say we shouldn't include it.

Comment (MikeMcQuaid) [reply Formula/archey4.rb:27]: Please remove this variable to move this pull request forward, thanks.

**Review by MikeMcQuaid — CHANGES_REQUESTED** [review]
Fine with this once variable is removed.

**Review by MikeMcQuaid — APPROVED** [review]

**Comment by MikeMcQuaid** [comment]
Thanks so much for your first contribution! Without people like you submitting PRs we couldn't run this project. You rock, @HorlogeSkynet!
